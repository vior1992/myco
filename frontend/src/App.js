import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { ToastProvider } from 'react-toast-notifications';
import Theme from './Theme';
import ContactsFactory from './components/ContactsFactory';
import ContactPage from './components/ContactPage';

function App() {
  return (
    <ToastProvider>
      <Theme>
        <Router>
          <Switch>
            <Route exact path="/">
              <ContactsFactory />
            </Route>
            <Route exact path="/contacts">
              <ContactPage />
            </Route>
            <Route path="/contacts/:id">
              <ContactPage />
            </Route>
          </Switch>
        </Router>
      </Theme>
    </ToastProvider>
  );
}

export default App;
