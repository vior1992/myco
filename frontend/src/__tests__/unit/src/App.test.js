import React from 'react';
import { render } from '@testing-library/react';
import App from '../../../App';

describe('App component', () => {
  it('should render without throwing an error & match snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
