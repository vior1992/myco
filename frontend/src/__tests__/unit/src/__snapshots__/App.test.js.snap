// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render without throwing an error & match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx VtiVB"
  >
    <div
      class="sc-gtssRu kJYrlK"
    >
      <h1>
        MYCO
        <span
          aria-label="monkey"
          role="img"
        >
          üêí
        </span>
      </h1>
      <h2>
        My Contact List
      </h2>
    </div>
    <a
      class="sc-dlnjPT bdksBB"
      href="/contacts"
    >
      New Contact
    </a>
    <div
      class="sc-hKFyIo jetHVG"
    />
  </div>
</DocumentFragment>
`;
